### PR TITLE
add pre commit hook for linting/formatting

### PR DESCRIPTION
**Pull request by estrattonbailey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -239,5 +243,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "lint-staged": {
+    "*": "yarn eslint --ext .js,.jsx,.ts,.tsx --fix"
```
Comment (estrattonbailey) [inline package.json:248]: `lint-staged` gets passed a list of all changed files like `<cmd> <file>[, <file>, ...]`, and I've found in the past it's a little easier to let respective CLIs handle exclusions or extensions than configure here in JSON — lmk if y'all disagree

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "1.41.0",
   "private": true,
   "scripts": {
+    "prepare": "is-ci || husky install",
```
Comment (estrattonbailey) [inline package.json:6]: `is-ci` prevents setting up git hooks in GitHub Actions

**Comment by pfrazee** [comment]
We couldn't get a +1 from @ansh on this so I'm inclined to close, wydt?

**Comment by estrattonbailey** [comment]
I think Ansh is willing to try it, but yeah I don't feel super strongly about it. Just seems to me like an obvious win to avoid [this kind of commit,](https://github.com/bluesky-social/social-app/pull/1012/commits/c6787bf1859ce5f6be4cd88cf2f4aa4722523e2c) but I could also set this up only for myself locally, too. Your call, no worries on my end!

**Review by ansh — APPROVED** [review]
LGTM 🚀

**Pull request merged**